scripts/globals/job_utils/dancer.lua

```diff
@@ -334,51 +411,46 @@ xi.job_utils.dancer.useViolentFlourishAbility = function(player, target, ability
     end
 end
 
-local waltzAbilities =
-{
---  Spell ID                           { tpCost, statMultiplier, baseHp }
-    [xi.jobAbility.CURING_WALTZ    ] = { 200, 0.25,  60 },
-    [xi.jobAbility.CURING_WALTZ_II ] = { 350, 0.50, 130 },
-    [xi.jobAbility.CURING_WALTZ_III] = { 500, 0.75, 270 },
-    [xi.jobAbility.CURING_WALTZ_IV ] = { 650, 1.00, 450 },
-    [xi.jobAbility.CURING_WALTZ_V  ] = { 800, 1.25, 600 },
-    [xi.jobAbility.DIVINE_WALTZ    ] = { 400, 0.25,  60 },
-    [xi.jobAbility.DIVINE_WALTZ_II ] = { 800, 0.75, 270 },
-}
-
-xi.job_utils.dancer.checkWaltzAbility = function(player, target, ability)
-    local waltzInfo = waltzAbilities[ability:getID()]
-
-    if target:getHP() == 0 then
-        return xi.msg.basic.CANNOT_ON_THAT_TARG, 0
-    elseif player:hasStatusEffect(xi.effect.SABER_DANCE) then
-        return xi.msg.basic.UNABLE_TO_USE_JA2, 0
-    elseif player:hasStatusEffect(xi.effect.TRANCE) then
-        ability:setRecast(math.min(ability:getRecast(), 6))
+xi.job_utils.dancer.useBuildingFlourishAbility = function(player, target, ability)
```

Comment:
This code is not valid with the current implementation of steps

Reply:
Code comented out, since it isn't correct. I will properly code it in a separate PR.